modules/widgets/simple-payments.php

```diff
@@ -278,10 +293,17 @@ public function validate_ajax_params( $params ) {
 				$errors->add( 'post_title', __( "People need to know what they're paying for! Please add a brief title.", 'jetpack' ) );
 			}
 
-			if ( empty( $params['price'] ) || floatval( $params['price'] ) <= 0 ) {
+			if ( empty( $params['price'] ) || ! is_numeric( $params['price'] ) || floatval( $params['price'] ) <= 0 ) {
 				$errors->add( 'price', __( 'Everything comes with a price tag these days. Please add a your product price.', 'jetpack' ) );
 			}
 
+			// Japan's Yen is the only supported currency with a zero decimal precision.
+			$precision = strcmp( $params['currency'], 'JPY' ) === 0 ? 0 : 2;
```

Comment:
Is there a reason why we're using `strcmp` instead of `===`?

Reply:
I think it's because it takes me back to the good old days of Visual Basic, when things were simpler...

> In those days spirits were brave, the stakes were high, men were real men, women were real women and small furry creatures from Alpha Centauri were real small furry creatures from Alpha Centauri.

Updated on 9e4df5d432b1f19107bcac26983f666f10773c77.